src/config/mod.rs

```diff
@@ -0,0 +1,127 @@
+//! Provides a simple way of reading configuration files.
+//!
+//! Reads a TOML config file for the IPC Agent and deserializes it in a type-safe way into a
+//! [`Config`] struct.
+
+mod deserialize;
+mod server;
+mod subnet;
+
+use std::fs;
+
+use anyhow::Result;
+use serde::Deserialize;
+pub use server::Server;
+pub use subnet::Subnet;
+pub use server::JSON_RPC_ENDPOINT;
+pub use crate::config::subnet::Subnets;
+
+pub const JSON_RPC_VERSION: &str = "2.0";
+
+/// The top-level struct representing the config. Calls to [`Config::from_file`] deserialize into
+/// this struct.
+#[derive(Deserialize)]
+pub(crate) struct Config {
+    pub server: Server,
+    pub subnets: Subnets,
+}
+
+impl Config {
+    /// Reads a TOML configuration in the `s` string and returns a [`Config`] struct.
+    pub fn from_str(s: &str) -> Result<Self> {
+        let config = toml::from_str(&s)?;
+        Ok(config)
+    }
+
+    /// Reads a TOML configuration file specified in the `path` and returns a [`Config`] struct.
+    pub fn from_file(path: &str) -> Result<Self> {
+        let contents = fs::read_to_string(path)?;
+        let config: Config = Config::from_str(contents.as_str())?;
+        Ok(config)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::net::SocketAddr;
+    use std::str::FromStr;
+
+    use fvm_shared::address::Address;
+    use indoc::formatdoc;
+    use ipc_sdk::subnet_id::{ROOTNET_ID, SubnetID};
+    use url::Url;
+
+    use crate::config::{Config, Server, Subnets};
+
+    // Arguments for the config's fields
+    const SERVER_JSON_RPC_ADDR: &str = "127.0.0.1:3030";
+    const ROOT_ID: &str = "/root";
+    const CHILD_ID: &str = "/root/f0100";
+    const ROOT_AUTH_TOKEN: &str = "ROOT_AUTH_TOKEN";
+    const CHILD_AUTH_TOKEN: &str = "CHILD_AUTH_TOKEN";
+    const JSONRPC_API_HTTP: &str = "https://example.org/rpc/v0";
+    const JSONRPC_API_WS: &str = "ws://example.org/rpc/v0";
+    const ACCOUNT_ADDRESS: &str = "f3thgjtvoi65yzdcoifgqh6utjbaod3ukidxrx34heu34d6avx6z7r5766t5jqt42a44ehzcnw3u5ehz47n42a";
+
+    fn check_server_config(config: &Server) {
```

Comment:
I think it would be preferable to have this function and `check_subnets_config` as `#[test]` instead of `read_config`. This way, if there are issues with a specific config section, we know immediately which one is it by looking at the test that is failing.

Reply:
Agree! Let's address this one and we should be good to go :pray:

Reply:
Cool, this is updated! Thx @adlrocha for the tip. Resolving this.